### PR TITLE
npm packages

### PR DESCRIPTION
**Pull request by turbocrime**
suggestions for @grod220 #672 

entry to vite-built packages is not complete, it renames them or something

### CODE REVIEW

File: config/tsconfig/base.json
Comment (turbocrime) [inline config/tsconfig/base.json:1]: see https://www.typescriptlang.org/docs/handbook/modules/reference.html#bundler

also, strictNullChecks is already enabled by strict

File: apps/extension/src/utils/rethrow-impl-errors.ts
Comment (turbocrime) [inline apps/extension/src/utils/rethrow-impl-errors.ts:1]: this file was weirdly reaching deep into connectrpc's cjs-specific exports to pull out some non-exported types. i just used an equivalent type - this isn't actually a refactor.

File: config/tailwind-config/package.json
Comment (turbocrime) [inline config/tailwind-config/package.json:1]: moved internal-only 'config' packages to a config dir, un-namespaced tailwind to match other configs

File: apps/node-status/package.json
Comment (turbocrime) [inline apps/node-status/package.json:1]: every package with a 'build' script now has a corresponding 'clean' script

File: packages/ui/package.json
Comment (turbocrime) [inline packages/ui/package.json:1]: this is only complex because the package is complex :/

**Comment by grod220** [comment]
Looks like our branches diverged quite a bit. Think it's best to keep `changset-npm` as the canonical branch, but will definitely consult this work as I go.

**Pull request closed without merging**